source/vir/src/ast_simplify.rs

```diff
@@ -330,6 +330,27 @@ fn simplify_one_expr(ctx: &GlobalCtx, state: &mut State, expr: &Expr) -> Result<
             let exprx = ExprX::Ctor(datatype, variant, binders, None);
             Ok(SpannedTyped::new(&expr.span, &expr.typ, exprx))
         }
+        ExprX::Assign { init_not_mut, lhs, rhs } => match &lhs.x {
+            ExprX::VarLoc(_) => Ok(expr.clone()),
+            ExprX::UnaryOpr(UnaryOpr::Field(_), _) => {
+                let (temp_decls, rhs) = small_or_temp(state, rhs);
```

Comment:
It looks like this evaluates `rhs` before `lhs`.  In the bigger picture, though, I think this code belongs in `ast_to_sst`, which is generally in charge of order of evaluation and introducing temporary variables.

Reply:
This should be address in https://github.com/secure-foundations/verus/pull/97/commits/bc7d2f3081e9305d25ff6a5281f61d896cbfda03